java/src/main/java/com/codurance/training/tasks/CheckAndUncheckServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package com.codurance.training.tasks;
+
+import java.io.BufferedReader;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Map;
+
+public class CheckAndUncheckServiceImpl implements CheckAndUncheckService{
+
+    private final Utility utility;
+    private final BufferedReader in;
+    private final PrintWriter out;
+    private final Map<String, List<Task>> tasks;
+
+    public CheckAndUncheckServiceImpl(BufferedReader in, PrintWriter out, Map<String, List<Task>> tasks) {
+        this.in = in;
+        this.out = out;
+        this.tasks = tasks;
+        this.utility = new Utility(this.in,this.out,this.tasks);
```

Comment:
make utility class methods public static and don't instantiate 
Better to have setDone as private within this class
Ref: http://blog.vanillajava.blog/2011/06/best-practice-dont-create-instances-of.html